Application/Frontend/components/Server/Member_List.tsx

```diff
@@ -1,35 +1,46 @@
 import React, { useEffect, useState } from 'react';
-import { Avatar, Group, Text, Stack, Paper, Button, Menu, rem } from '@mantine/core';
+import { Avatar, Group, Text, Stack, Paper, Button, Menu, rem, TextInput, Modal } from '@mantine/core';
 import { IconSettings, IconMessageCircle, IconPhoto, IconSearch, IconArrowsLeftRight, IconTrash, IconPlus, IconUserUp } from '@tabler/icons-react';
 import { useUser } from '@clerk/nextjs';
+import { channel } from 'diagnostics_channel';
+import { createHash } from 'crypto';
+import { IntegerType } from 'mongodb';
+import { useDisclosure } from '@mantine/hooks';
+import { showNotification } from '@mantine/notifications';
 
-export function MemberList({isAdmin}: any) {
+const generateHash = (input: string) => {
+  return createHash('sha256').update(input).digest('hex');
+};
+
+export function MemberList({isAdmin, chatID}: any) {
   // Hardcoded member list
   const [membersList, setMembersList] = useState<string[]>([]);
-  const [serverID, setServerID] = useState<string>("2");
+  const [membersIDList, setMembersIDList] = useState<string[]>([]);
+  // const [membersIDListToSort, setMembersIDListToSort] = useState<string[]>([]);
+  const [channelKey, setChannelKey] = useState<string>(chatID);
+  const [opened, { open, close }] = useDisclosure(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [friendUsername, setFriendUsername] = useState('');
+  const [searchResult, setSearchResult] = useState<number | null>(null);
   const { user } = useUser();
 
-//   function handleButtonClick(member: String) {
-//     try {
-//         // Your asynchronous operation here
-//         removeMember(member);
-//         console.log('Async operation completed');
-//     } catch (error) {
-//         console.error('Error:', error);
-//     }
-// }
-
-  const removeMember = async(member: String) =>{
-      console.log("Something")
-      alert("Something")
+
+  const removeMember = async(member: String, index: any) =>{
+      console.log(membersIDList[index] + " ahahahahahah")
```

Comment:
Good use of `console.log` statements for debugging, but be sure to remove these before merging.